config/e2guardian/e2guardian.inc

```diff
@@ -0,0 +1,1458 @@
+				file_put_contents("/tmp/cert.pem",$crt, LOCK_EX);
+				$cert_hash=array();
+				exec("/usr/bin/openssl x509 -hash -noout -in /tmp/cert.pem",$cert_hash);
+				file_put_contents("/usr/local/share/certs/".$cert_hash[0].".0",$crt,LOCK_EX);
+				$crt="";
+				$cert=0;
+				}
+			}
+		}
+}
+function sync_package_e2guardian($via_rpc="no",$install_process=false) {
+	global $config,$g;
+
+	# detect boot process
+	if (is_array($_POST)){
+		if (preg_match("/\w+/",$_POST['__csrf_magic']))
+			unset($boot_process);
+		else
+			$boot_process="on";
+	}
+
+	if (is_process_running('e2guardian') && isset($boot_process) && $via_rpc=="no"){
+		log_error("[E2guardian] - Detected boot process pr:".is_process_running('e2guardian')." bp:".isset($boot_process)." rpc:".$via_rpc);
+		return;
+		}
+	else{
+		log_error("[E2guardian] - Save settings package call pr:".is_process_running('e2guardian')." bp:".isset($boot_process)." rpc:".$via_rpc);
+	}
+
+	if (isset($_POST["apply"])){
+		e2guardian_start($via_rpc,$install_process);
+		unlink_if_exists(E2GUARDIAN_DIRTYPATH);
+		return;
+	}
+
+	#assign xml arrays
+	if (!is_array($config['installedpackages']['e2guardian']))
+		$config['installedpackages']['e2guardian']['config'][0]=array('interface'=>'lo0',
+																		'daemon_options'=>'softrestart');
+	$e2guardian=$config['installedpackages']['e2guardian']['config'][0];
+	if (!is_array($config['installedpackages']['e2guardianconfig']))
+		$config['installedpackages']['e2guardianconfig']['config'][0]=array('auth_plugin'=>'',
+																		'scan_options'=>'scancleancache,createlistcachefiles,deletedownloadedtempfiles',
+																		'weightedphrasemode'=>'2',
+																		'preservecase'=>'0',
+																		'phrasefiltermode'=>'2',
+																		'cron'=>'day');
+	$e2guardian_config=$config['installedpackages']['e2guardianconfig']['config'][0];
+	if (!is_array($config['installedpackages']['e2guardianlog']))
+		$config['installedpackages']['e2guardianlog']['config'][0]=array('report_level'=>'3',
+																		'report_language'=>'ukenglish',
+																		'report_options'=>'showweightedfound,usecustombannedimage,nonstandarddelimiter',
+																		'logging_options'=>'logconnectionhandlingerrors',
+																		'loglevel'=>'2',
+																		'logexceptionhits'=>'2',
+																		'logfileformat'=>'1');
+	$e2guardian_log=$config['installedpackages']['e2guardianlog']['config'][0];
+	if (is_array($config['installedpackages']['e2guardianlimits']))
+		$e2guardian_limits=$config['installedpackages']['e2guardianlimits']['config'][0];
+	if (is_array($config['installedpackages']['e2guardianusers']))
+		$e2guardian_users=$config['installedpackages']['e2guardianusers']['config'][0];
+	if (is_array($config['installedpackages']['e2guardianblacklist']['config']))
+		$e2guardian_blacklist=$config['installedpackages']['e2guardianblacklist']['config'][0];
+
+	#daemon options
+	$e2guardian_enabled=$e2guardian['enable_dg'];
```

Comment:
This seems odd - and I don't see enable_dg or $e2guardian_enabled anywhere else here.